llms/mlx_lm/tuner/trainer.py

```diff
@@ -92,6 +92,9 @@ def iterate_batches(dataset, tokenizer, batch_size, max_seq_length, train=False)
         for i in indices:
             # Encode batch
             batch = [tokenizer.encode(dataset[j]) for j in batch_idx[i]]
+            for b in batch:
+                b.append(tokenizer.eos_token_id)
+
```

Comment:
I was trying to figure out if there's some something in the tokenizer's interface that let's you control this but it seems it could be different across [all supported ones](https://huggingface.co/docs/transformers/en/model_doc/auto#transformers.AutoTokenizer)?

I guess if we wanted to be really defensive we could check if the last token of the sequence is already `eos_token_id`? Maybe overkill though.

Reply:
I spent a long time trying to figure that out myself and did not succeed so I gave up and wrote this hack 😅 

It's a good idea to check the end. I was wondering myself.. if it ever happens. I don't think it does because then presumably our generation code would be broken (we'd be adding eos to the prompt) but who knows.